lib/actions/help.js

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2016-2020 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as _ from 'lodash';
+
+import * as capitano from 'capitano';
+import * as columnify from 'columnify';
+import * as messages from '../utils/messages';
+import { getManualSortCompareFunction } from '../utils/helpers';
+import { exitWithExpectedError } from '../utils/patterns';
+import { getOclifHelpLinePairs } from './help_ts';
+
+const parse = object =>
+	_.map(object, function(item) {
+		// Hacky way to determine if an object is
+		// a function or a command
+		let signature;
+		if (item.alias != null) {
+			signature = item.toString();
+		} else {
+			signature = item.signature.toString();
+		}
+
+		return [signature, item.description];
+	});
+
+const indent = function(text) {
+	text = _.map(text.split('\n'), line => '    ' + line);
+	return text.join('\n');
+};
+
+const print = usageDescriptionPairs =>
+	console.log(
+		indent(
+			columnify(_.fromPairs(usageDescriptionPairs), {
+				showHeaders: false,
+				minWidth: 35,
+			}),
+		),
+	);
+
+const manuallySortedPrimaryCommands = [
+	'help',
+	'login',
+	'push',
+	'logs',
+	'ssh',
+	'apps',
+	'app',
+	'devices',
+	'device',
+	'tunnel',
+	'preload',
+	'build',
+	'deploy',
+	'join',
+	'leave',
+	'local scan',
+];
+
+const general = function(_params, options, done) {
+	console.log('Usage: balena [COMMAND] [OPTIONS]\n');
+	console.log(messages.reachingOut);
+	console.log('\nPrimary commands:\n');
+
+	// We do not want the wildcard command
+	// to be printed in the help screen.
+	const commands = capitano.state.commands.filter(
+		command => !command.hidden && !command.isWildcard(),
+	);
```

Comment:
Changed this from
```ts
	const commands = _.reject(
		capitano.state.commands,
		command => command.hidden || command.isWildcard(),
	);
```
to this native version as the native version works properly with types